paddle/fluid/operators/detail/grpc_server.cc

```diff
@@ -337,27 +383,37 @@ void AsyncGRPCServer::HandleRequest(::grpc::ServerCompletionQueue* cq,
       VLOG(3) << "HandleRequest for " << cq_name << " after WaitCond";
     }
 
-    RequestBase* base = reinterpret_cast<RequestBase*>(tag);
+    RequestBase* base = nullptr;
+    {
+      std::lock_guard<std::mutex> l(cq_mutex_);
+      if (cq_name == "cq_get") {
+        base = get_reqs_[i];
+      } else if (cq_name == "cq_send") {
+        base = send_reqs_[i];
+      } else {
+        CHECK(false);
+      }
+    }
     // reference:
     // https://github.com/tensorflow/tensorflow/issues/5596
     // https://groups.google.com/forum/#!topic/grpc-io/xftlRy-IQwM
     // https://groups.google.com/forum/#!topic/grpc-io/ywATt88Ef_I
     if (!ok) {
       LOG(WARNING) << cq_name << " recv no regular event:argument name["
                    << base->GetReqName() << "]";
-      TryToRegisterNewOne();
+      TryToRegisterNewOne(i);
```

Comment:
Why we need `send_reqs_ ` `get_reqs_ ` since elements in it can't be reusable?

Reply:
I'm just following the grpc stress test example. I suspect pre-allocating them could improve performance (as the benchmark shows)